flow/aggregates.py

```diff
@@ -0,0 +1,505 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from collections.abc import Iterable
+from hashlib import sha1
+from itertools import groupby
+from itertools import zip_longest
+
+
+class aggregator:
+    r"""Decorator for operation function that is to be aggregated.
+    By default, if the ``aggregator_function`` is not passed,
+    an aggregate of all jobs will be created.
+
+    .. code-block:: python
+
+        @aggregator()
+        @FlowProject.operation
+        def foo(*jobs):
+            print(len(jobs))
+
+    :param aggregator_function:
+        Information describing how to aggregate jobs. Is a callable that
+        takes in a list of jobs and can return or yield subsets of jobs as
+        an iterable. The default behavior is creating a single aggregate
+        of all jobs.
+    :type aggregator_function:
+        callable
+    :param sort_by:
+        Before aggregating, sort the jobs by a given statepoint parameter.
+        The default behavior is no sorting.
+    :type sort_by:
+        str or NoneType
+    :param reverse_order:
+        States if the jobs are to be sorted in reverse order.
+        The default value is False.
+    :type reverse_order:
+        bool
+    :param select:
+        Condition for filtering individual jobs. This is passed as the
+        callable argument to `filter`.
+        The default behavior is no filtering.
+    :type select:
+        callable or NoneType
+    :param \*\*kwargs:
+        Additional information related to aggregator_function function.
```

Comment:
Why are we accepting `kwargs` when they are not used anywhere meaningfully?

Reply:
These are used to check equality between two aggregator objects. I was having difficulty in comparing aggregators hence I thought that these additional parameters would be helpful to test the equality

Reply:
If this is just for testing equality, please remove it. This feature is not used for any purpose. We should deal directly with the problem of testing equality. What part was giving you difficulty?